src/Symfony/Bundle/FrameworkBundle/Controller/ControllerResolver.php

```diff
@@ -15,17 +15,31 @@
 use Symfony\Component\DependencyInjection\ContainerAwareInterface;
 use Symfony\Component\DependencyInjection\ContainerInterface;
 use Symfony\Component\HttpKernel\Controller\ContainerControllerResolver;
+use Symfony\Component\HttpKernel\KernelInterface;
 
 /**
  * @author Fabien Potencier <fabien@symfony.com>
+ *
+ * @final since Symfony 4.4
  */
 class ControllerResolver extends ContainerControllerResolver
 {
+    /**
+     * @deprecated since Symfony 4.4
+     */
     protected $parser;
 
-    public function __construct(ContainerInterface $container, ControllerNameParser $parser, LoggerInterface $logger = null)
+    public function __construct(ContainerInterface $container, /* LoggerInterface */ $logger = null)
```

Comment:
let's add a docblock instead of the inline comment?

Reply:
done